### ISSUE TITLE
Premailer not working with devise mails.

### ISSUE THREAD

**Issue by AvanthikaMeenakshi** [premailer/premailer#312]
In the inky + premailer combination with devise mails, i think even before the mails are rendered, inlining is getting completed. Other mails are working with inky.rb+premailer, but inlining just doesnt happen for users/mailer even though inky is perfectly rendering the table layout.

**Comment by TechCanuck** [comment]
Agreed, they tend to get appended as attachments and the yeild isn't getting respected with the inline.   Anytime the .inky is attached to inline code (ie., user_mailer/reset_password_instructions.html.erb) it just shows as the body of hte email, and any inky layout is attached as an attachment.

**Comment by simoniong** [comment]
I think i have the same issue here, anyone got a fix for this?

**Comment by ashwinv11** [comment]
I was able to fix this and use inky by overriding the `Devise::Mailer` class and using my own class:

In `app/mailers/devise_mailer.rb`:
```
class DeviseMailer < Devise::Mailer
  helper :application
  include Devise::Controllers::UrlHelpers
  default template_path: 'devise/mailer'
  layout 'mailer' # I actually have a layouts/mailer.html.inky file I'm using for the layout as well
end
```

Then in `config/initializers/devise.rb` I added:
```
...
config.mailer = 'DeviseMailer'
...
```
Finally, I overrode the views like so: `views/devise/mailer/confirmation_instructions.html.inky`, etc.

The styles were getting rendered correctly, but I had issues with the Rails `image_tag` helper not loading images inline, as well as fonts not rendering due to incompatibility in email clients. I fixed these issues by:

1. Adding the asset_host line to my `application.rb`: 
```
...
config.action_mailer.asset_host = 'http://example.com'
...
```
Be mindful, that this can be changed on a per enviornment basis. 

2. Adding appropriate fallback fonts

A bonus tip: turns out Gmail doesn't like SVGs, so I switched out my images for PNGs. Hope that helps!
